src/raft.c

```diff
@@ -341,23 +361,43 @@ void RaftExecuteCommandArray(RedisRaftCtx *rr,
         int old_entered_eval = rr->entered_eval;
 
         if ((cmdlen == 4 && !strncasecmp(cmd, "eval", 4)) ||
-            (cmdlen == 7 && !strncasecmp(cmd, "evalsha", 7))) {
+            (cmdlen == 7 && !strncasecmp(cmd, "evalsha", 7)) ||
+            (cmdlen == 5 && !strncasecmp(cmd, "fcall", 5))) {
```

Comment:
what about `X_ro` cmds?

Reply:
ok, yea, this is a bit old, I'm wondering if we it be worth to do something like

```
        if (((cmdlen == 4 || cmdlen == 7 || cmdlen == 10) && !strncasecmp(cmd, "eval", 4)) ||
            ((cmdlen == 5 || cmdlen == 8) && !strncasecmp(cmd, "fcall", 5))) {
            rr->entered_eval = 1;
        }
```

i.e. 4 is eval, 7 is evalsha and eval_ro and 10 is evalsha_ro and the same for fcall.

Reply:
not neccessary